react/src/components/ComputeSessionNodeItems/SessionStatusTag.tsx

```diff
@@ -0,0 +1,109 @@
+import Flex from '../Flex';
+import {
+  SessionStatusTagFragment$data,
+  SessionStatusTagFragment$key,
+} from './__generated__/SessionStatusTagFragment.graphql';
+import { LoadingOutlined } from '@ant-design/icons';
+import { Tag, theme } from 'antd';
+import graphql from 'babel-plugin-relay/macro';
+import _ from 'lodash';
+import React from 'react';
+import { useFragment } from 'react-relay';
+
+interface SessionStatusTagProps {
+  sessionFrgmt?: SessionStatusTagFragment$key | null;
+}
+const statusTagColor = {
+  //prepare
+  RESTARTING: 'blue',
+  PREPARING: 'blue',
+  PULLING: 'blue',
+  //running
+  RUNNING: 'green',
+  PENDING: 'green',
+  SCHEDULED: 'green',
+  //error
+  ERROR: 'red',
+  //finished return undefined
+};
+
+const isTransitional = (session: SessionStatusTagFragment$data) => {
+  return [
+    'RESTARTING',
+    'TERMINATING',
+    'PENDING',
+    'PREPARING',
+    'PULLING',
+  ].includes(session?.status || '');
+};
+
+const statusInfoTagColor = {
+  // 'idle-timeout': undefined,
+  // 'user-requested': undefined,
+  // scheduled: undefined,
+  // 'self-terminated': undefined,
+  'no-available-instances': 'red',
+  'failed-to-start': 'red',
+  'creation-failed': 'red',
+};
+const SessionStatusTag: React.FC<SessionStatusTagProps> = ({
+  sessionFrgmt,
+}) => {
+  const session = useFragment(
+    graphql`
+      fragment SessionStatusTagFragment on ComputeSessionNode {
+        id
+        name
+        status
+        status_info
+      }
+    `,
+    sessionFrgmt,
+  );
+  const { token } = theme.useToken();
+
+  return session ? (
+    _.isEmpty(session.status_info) ? (
+      <Tag
+        color={
+          session.status ? _.get(statusTagColor, session.status) : undefined
+        }
+        icon={isTransitional(session) ? <LoadingOutlined spin /> : undefined}
```

Comment:
Even if session terminating is completed, the status still `TERMINATING`.

Reply:
Yes, currently there is no auto-refresh. This can be resolved after the session list is implemented using the `ComputeSessionNode` type. After that, the WebUI will periodically check the updated active session list using GraphQL, and then the places where fragments are used will also be automatically updated.

Nevertheless, it may be a degradation in UX, so I will consider implementing auto-refresh in the session detail panel. If necessary, I will proceed with it as a separate stack, so please review this PR again.